devtools/working-with-dsproxy/working-with-dsproxy.md

```diff
@@ -0,0 +1,488 @@
+
+Scripts are implemented in Solidity as functions and multiple scripts are typically combined and deployed together as a single contract. A DSProxy contract can only execute one script in a single transaction. In this section we will focus on the features of a DSProxy contract and look at how scripts work in the next section.
+
+#### Ownership
+
+Ownership of a DSProxy contract is set to an address when it is deployed. There is support for authorities based on DSAuth if there is a need for ownership of the DSProxy contract to be shared among multiple users.
+
+#### Execute
+
+`execute(address target, bytes data)` function implements the core functionality of DSProxy. It takes in two inputs, an `address` of the contract containing scripts, and `data` which contains calldata to identify the script that needs to be executed along with it's input data.
+
+`msg.sender` when the script is being executed will continue to be the user address instead of the address of the DSProxy contract.
+
+`execute(bytes code, bytes data)` is an additional function that can be used when a user wants to deploy a contract containing scripts and then call one of the scripts in a single transaction. A `cache` registers the address of contract deployed to save gas by skipping deployment when other users call `execute` with the same bytecode later.
+
+#### Event Logs
+
+A DSProxy contract generates a event called `LogNote` with these values indexed when `execute()` is called,
+
+* Function signature, `0x1cff79cd`
+* Owner of the DSProxy contract, `msg.sender`
+* Contract address which contains the script, `address`
+* Calldata which contains function signature of script being executed and its input data, `data`
+
+#### Factory Contract
+
+The function  `build` in the DSProxyFactory contract is used to deploy a personal DSProxy contract. For production usecases on mainnet you can use a common factory contract that is already being used by existing projects to avoid deploying redundant DSProxy contracts for users who already have one. Please check the [Production Usage](/production-usage) section in this guide for more details.
+
+### Create a script
+
+We've seen an example earlier of how a script can help CDP owners pay back their Dai debt by purchasing the required amount of MKR from an exchange within the same transaction. Uniswap exchange contracts are a good source of liquidity especially for buying small amounts of MKR. In this section, we will create a script that will allow users to buy MKR with DAI from Uniswap and wipe debt from a CDP.
+
+#### Environment Setup
+
+We'll use `dapp` and `seth` while working through this section but you can also your own tool of choice like the Remix IDE to execute these steps. Instructions to install both the tools can be found [here](https://dapp.tools/).
+
+You have to create a `~/.sethrc` file and configure it with these values to work with the Kovan testnet,
+
+* `SETH_CHAIN=kovan`
```

Comment:
Spent like 30 min trying to understand why ~/.sethrc wasn't running/being checked... I realized I missed the `export` command before every line! haha!  Could we add these commands to every config line? Such as `export SETH_CHAIN=kovan`, `export ETH_FROM ... `, etc?